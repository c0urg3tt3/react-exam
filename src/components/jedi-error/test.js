import React from 'react'
import { shallow } from 'enzyme'

import JediError, {
  BOUNDARY_ERROR_QUOTE,
  FETCHING_ERROR_QUOTE
} from './component'

describe('[component] JediError', () => {
  it('should render component', () => {
    const wrapper = shallow(
      <JediError>
      {
        ({message, quote, autor}) => (
          <div>
            <p className="message">{message}</p>
            <blockquote>
              <p className="quote">{quote}</p>
              <footer className="autor">{autor}</footer>
            </blockquote>
          </div>
        )
      }
      </JediError>
    )

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(false)
    expect(wrapper.hasClass('.jedi-error')).toEqual(false)

    const errorMessage = "No, I am your father!"
    wrapper.setProps({
      errorMessage: errorMessage,
      isBoundaryError: true
    })

    expect(wrapper.find('.message').text()).toEqual(errorMessage)
    expect(wrapper.find('.quote').text()).toEqual(BOUNDARY_ERROR_QUOTE.quote)
    expect(wrapper.find('.autor').text()).toEqual(BOUNDARY_ERROR_QUOTE.autor)

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({
      isBoundaryError: false,
      isFetchingError: true
    })

    expect(wrapper.find('.message').text()).toEqual(errorMessage)
    expect(wrapper.find('.quote').text()).toEqual(FETCHING_ERROR_QUOTE.quote)
    expect(wrapper.find('.autor').text()).toEqual(FETCHING_ERROR_QUOTE.autor)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
