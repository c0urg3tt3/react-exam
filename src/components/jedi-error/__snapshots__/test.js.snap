// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediError should render component when boundary error 1`] = `"<div class=\\"jedi-error\\"><div><p class=\\"message\\">No, I am your father!</p><blockquote><p class=\\"quote\\">I felt a great disturbance in the Force, as if millions of voices suddenly cried out in terror and were suddenly silenced. I fear something terrible has happened.</p><footer class=\\"autor\\">Obi-Wan Kenobi</footer></blockquote></div></div>"`;

exports[`[component] JediError should render component when fetch error 1`] = `"<div class=\\"jedi-error\\"><div><p class=\\"message\\">No, I am your father!</p><blockquote><p class=\\"quote\\">What if I told you that the Republic was now under the control of a Dark Lord of the Sith?</p><footer class=\\"autor\\">Count Dooku</footer></blockquote></div></div>"`;

exports[`[component] JediError should render component when post error 1`] = `"<div class=\\"jedi-error\\"><div><p class=\\"message\\">No, I am your father!</p><blockquote><p class=\\"quote\\">The Council will act as they deem necessary.</p><footer class=\\"autor\\">Mace Windu, member of the Jedi High Council</footer></blockquote></div></div>"`;

exports[`[component] JediError should render empty component 1`] = `null`;
